### DISCUSSION TITLE
How to expand HTML tag by class name or id

### DISCUSSION THREAD

**Discussion by orklann** [dcampos/nvim-snippy#105] [General]
Hello, great plugin.

Here is my question: How can I expand HTML tags like this:
`.topic` to expand as `<div class="topic"></div>`
`#name` to expand as `<div id="name"></div>`

How to write these two kinds of snippets?

**Comment by dcampos** [comment]
Hi! Snippy currently doesn't support regex-based triggers, although other snippet managers, like Ultisnips, do.

Another option is to use [emmet-vim](https://github.com/mattn/emmet-vim), which supports expanding `div.topic` to `<div class="topic"></div>` and much more. It has its own set of mappings to expand or jump around, so it might take some time to get used to it. I've written a [plugin](https://github.com/dcampos/cmp-emmet-vim) that allows you to complete emmet snippets with nvim-cmp, so they work just like normal snippets.

There is also [emmet-ls](https://github.com/aca/emmet-ls), which you might find useful.

**Comment by orklann** [comment]
Thank you very much for the suggestion on alternatives. I will keep using this plugin, it's so easy to use.
